packages/flutter_tools/lib/src/cache.dart

```diff
@@ -1469,6 +1469,14 @@ class ArtifactUpdater {
       try {
         _ensureExists(tempFile.parent);
         await _net.fetchUrl(url, destFile: tempFile, maxAttempts: 2);
+      } on Exception {
```

Comment:
Ehhhh. Okay, sorry. Maybe that wasn't the best suggestion. If there's some way to avoid catching `Exception` here that would be good, though, since it's only the `IOExceptions` that probably indicate a network issue on the end-user's side.